.editorconfig

```diff
@@ -0,0 +1,3 @@
+[*.{yaml,yml}]
```

Comment:
Can you take away the `.editorconfig` into its own commit?

It would be useful to add rules for the other usual suspects we have: Go files, Markdown files, Python files, shell scripts, Dockerfiles, etc.  

And you can also add a general section about trimming trailing white space, inserting final new line, etc.

For inspiration, please see what our Invenio/Indico/Zenodo colleagues use:

- https://github.com/inveniosoftware/invenio-records/blob/master/.editorconfig
- https://github.com/indico/indico/blob/master/.editorconfig
- https://github.com/zenodo/zenodo/blob/master/.editorconfig

We should however modernise the rules, e.g. for Python we are using `black` code formatting, so the `.editorconfig` rule might look as simply as:

```
[*.py]
profile = black
```

So, can you please look around at the best practices, pick a configuration that is very close to our current code formatting in nature (so that our code base would not have to change much), and assemble all these editor rules into a new separate commit in order to make a separate release note announcement? (Perhaps under the `style` scope of the conventional commits fame, see https://github.com/reanahub/reana/wiki/Tips-for-Git)